runtime/near-vm-runner/src/instrument/stack_height/mod.rs

```diff
@@ -90,6 +90,7 @@ mod thunk;
 ///
 /// This means that the module is invalid.
 #[derive(Debug)]
+#[allow(dead_code)]
```

Comment:
Was this complaining about the field within the struct here (and elsewhere?) Is there a more granular lint we can allow than `dead_code`? Cause this I believe will keep this struct (or enum) around even after it becomes truly unused.

Reply:
Complain [here](https://github.com/rust-lang/rust/pull/85200) I guess. ;)

But yeah, I’m not happy with this either.  The best I can think of is to declare a dummy function with `allow(dead_code)` which accesses those fields.

Reply:
Turns out types in chain/client/src/adapter.rs weren’t used and Clippy failed to detect that for some reason.  Does derive marks type as used perhaps? In which case `allow(dead_code)` won’t prevent unused type detection.

Reply:
> Turns out types in chain/client/src/adapter.rs weren’t used and Clippy failed to detect that for some reason. Does derive marks type as used perhaps?

Might very well be some of the derives creating a use that's not obvious. e.g. maybe they put the type id into some sort of type registry?

---

> Complain https://github.com/rust-lang/rust/pull/85200 I guess. ;)

Ah, I see. I actually agree with the reasoning in the PR and would remove the field here.